felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -625,6 +645,35 @@ func (m *bpfEndpointManager) onProfileRemove(msg *proto.ActiveProfileRemove) {
 	m.markEndpointsDirty(m.profilesToWorkloads[profID], "profile")
 	delete(m.profiles, profID)
 	delete(m.profilesToWorkloads, profID)
+	if m.bpfPolicyDebugEnabled {
+		m.polNameToRuleIDs[profID.Name] = markAllRuleIdsDirty(m.polNameToRuleIDs[profID.Name])
+	}
+}
+
+func (m *bpfEndpointManager) removeDirtyPolicies() {
+	for k, v := range m.polNameToRuleIDs {
+		v.Iter(func(item bpfRuleInfo) error {
+			if item.dirty {
+				if val, ok := m.ruleIdToMatchID[item.ruleId]; ok {
+					b := make([]byte, 8)
```

Comment:
do we want to keep allocating the memory? We can reuse the same array for all iterations, just write the value in it.